File: app/src/main/java/com/idax/openvision/Room/AppDatabase.java
package com.idax.openvision.Room;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.idax.openvision.Entity.User;

@Database(entities = {User.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    public static AppDatabase instance = null;

    public static AppDatabase getAppDataBase(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context, AppDatabase.class, "open-vision.db").allowMainThreadQueries().addCallback(new RoomDatabase.Callback() {
                @Override
                public void onCreate(@NonNull SupportSQLiteDatabase db) {
                    super.onCreate(db);
                }
            }).build();
        }
        return instance;
    }

    public abstract UserDao userDao();
}
/*
@Database(entities = {Obj.class, Etc.class}, version = 1, exportSchema = false)
	public abstract class AppDataBase extends RoomDatabase {
		public static AppDataBase instance = null;
		public static AppDataBase getAppDataBase(Context context) {
			if (instance == null) {
				instance = Room.databaseBuilder(context, AppDataBase.class, "inventory.db").allowMainThreadQueries().addCallback(new RoomDatabase.Callback() {
					@Override
					public void onCreate(@NonNull SupportSQLiteDatabase db) {
						super.onCreate(db);
					}
				}).build();
			}
			return instance;
		}
 */

File: app/src/main/java/com/idax/openvision/Room/AppDatabase_.kt
package com.idax.openvision.Room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.idax.openvision.Entity.User


@Database(entities = arrayOf(User::class), version = 1, exportSchema = false)
abstract class AppDatabase_ : RoomDatabase() {

    companion object {
        @Volatile private var instance: AppDatabase_? = null
        private val LOCK = Any()

        operator fun invoke(context: Context)= instance ?: synchronized(LOCK){
            instance ?: buildDatabase(context).also { instance = it}
        }

        private fun buildDatabase(context: Context) = Room.databaseBuilder(context,
            AppDatabase_::class.java, "open-vision.db")
            .build()
    }

    abstract fun userDao(): UserDao
}